### PR TITLE
MINOR: create KafkaConfigSchema and TimelineObject

### PR DESCRIPTION
**Pull request by cmccabe**
Create KafkaConfigSchema to encapsulate the concept of determining the types of configuration keys. This is useful in the controller because we can't import KafkaConfig, which is part of core. Also introduce the TimelineObject class, which is a more generic version of TimelineInteger / TimelineLong.

### CODE REVIEW

File: metadata/src/main/java/org/apache/kafka/metadata/KafkaConfigSchema.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.ConfigResource;
+import org.apache.kafka.common.metadata.ConfigRecord;
+
+import java.util.Map;
+
+import static java.util.Collections.emptyMap;
+
+
+public class KafkaConfigSchema {
```
Comment (mumrah) [inline metadata/src/main/java/org/apache/kafka/metadata/KafkaConfigSchema.java:34]: Can we add a brief javadoc explaining the purpose of this class?

File: core/src/main/scala/kafka/server/KafkaRaftServer.scala [hunk 2]
```diff
@@ -176,4 +180,8 @@ object KafkaRaftServer {
     (metaProperties, offlineDirs.toSeq)
   }
 
+  val configSchema = new KafkaConfigSchema(Map(
+    ConfigResource.Type.BROKER -> KafkaConfig.configDef,
+    ConfigResource.Type.TOPIC -> LogConfig.configDefCopy,
```
Comment (mumrah) [inline core/src/main/scala/kafka/server/KafkaRaftServer.scala]: I realize this is just copied over from ControllerServer, but why do we make a copy of the LogConfig but not KafkaConfig here?

**Review by mumrah — APPROVED** [review]
Thanks, @cmccabe! It's nice to see the ConfigDef code moved out of ConfigurationControlManager. I think this change looks good. Two non-critical comments inline

File: core/src/main/scala/kafka/server/KafkaRaftServer.scala (hunk 2 above)
Comment (cmccabe) [reply core/src/main/scala/kafka/server/KafkaRaftServer.scala]: In practice nobody mutates either of these. But you're right, it is inconsistent. I will do a defensive copy on the KafkaConfig one.

In the future we should have a freeze method or something on the ConfigDef to make these defensive copies unnecessary. But that's out of scope here...

**Pull request merged**